models/topics.models.js

```diff
@@ -4,20 +4,33 @@ const fs = require("fs/promises")
 function fetchTopics(){
     return db.query('SELECT * FROM topics').then(({rows}) => {
         return rows;
-    }).catch((err) => {
-        next(err);
-    });
+    })
 };
 
 function fetchApi(){
     return fs.readFile("./endpoints.json", 'utf-8')
     .then((data) => {
         return JSON.parse(data)
-    }).catch((err) => {
-        next(err)
-    }); 
+    })
 }
 
+function fetchArticleById(article_id, next){
+    const query = ` SELECT * FROM articles
+                    WHERE articles.article_id = $1;`
+   
+    return db.query(query, [article_id])
+        .then((result) => {
+            if (result.rows.length === 0){
+                return Promise.reject({status: 404, msg: 'article_id does not exist'})
+            } else {
+                return result.rows[0];
+            }
+        })
+        .catch((err) => {
```

Comment:
We shouldn't have multiple catch blocks within our promise chain, lets just stick to having the one in our controller and let that handle any errors thrown in the chain and pass them to our middleware.